src/ast/irbuilderbpf.cpp

```diff
@@ -475,24 +475,61 @@ CallInst *IRBuilderBPF::CreateGetStackScratchMap(StackType stack_type,
                              failure_callback);
 }
 
-Value *IRBuilderBPF::CreateGetStrScratchBuffer(const location &loc, int key)
-{
-  return createScratchBuffer(bpftrace::globalvars::GlobalVar::GET_STR_BUFFER,
-                             loc,
-                             key);
+Value *IRBuilderBPF::CreateGetStrStackOrScratchBufferAllocation(
```

Comment:
oofa these function names - what is this Java?. Is something like `CreateGetStrAllocation` not accurate or descriptive enough?

Reply:
lol yes these are too wordy...i debated changing it and I should've. Yes lets go with Allocation